src/order/well_founded.lean

```diff
@@ -20,6 +20,25 @@ and provide a few new definitions: `well_founded.min`, `well_founded.sup`, and `
 variables {α : Type*}
 
 namespace well_founded
+
+theorem not_gt_of_lt {α : Sort*} {r : α → α → Prop} (h : well_founded r) :
```

Comment:
You might be surprised to find out that if you write `{α : Sort*}` in a lemma, lean will sometimes be happy to silently convert it to `{α : Type*}`. If you put it in a `variables` it won't do that.

What you have here means the right thing fortunately.